tagsets/expregister.php

```diff
@@ -215,11 +215,15 @@ function expregister__get_history($participant_id) {
 }
 
 function expregister__list_history($participant) {
-    global $lang, $color, $preloaded_laboratories;
+    global $settings, $lang, $color, $preloaded_laboratories, $preloaded_payment_types;
 
     if (!(is_array($preloaded_laboratories) && count($preloaded_laboratories)>0))
         $preloaded_laboratories=laboratories__get_laboratories();
-
+        
+    if (!(is_array($preloaded_payment_types) && count($preloaded_payment_types)>0)) {
```

Comment:
Likewise, `empty` will work on arrays too to make this line shorter.
```suggestion
    if (empty($preloaded_payment_types)) {
```

Reply:
But thinking more, why you need to rely on global and not just `$preloaded_payment_types=payments__load_paytypes();` call within this function?

Reply:
Yes, I am a little verbatim sometimes. I somehow find it cleaner to test existence and value directly than to rely on an opaque function. But yes, that would be shorter.

The idea behind the global preloads is to save database queries and improve performance. It might not necessarily apply in this specific case, but generally many of these queries loading lists of items may be executed many times on a single page. So I systematically save them to globals and check whether they already have been loaded.

Reply:
I see, thanks for explanation.